### PR TITLE
Chore/upgrade cypress 12

### PR DESCRIPTION
**Pull request by Isalafont**
⚠️ Running the cypress test created enrollments in local dev environnement ! 

since commit 2e7ec451bfce1ce7ce93e74da393c5c99c27c63c

### CODE REVIEW

File: .github/workflows/end-to-end.yml [hunk 1]
```diff
@@ -62,6 +64,7 @@ jobs:
         with:
           path: backend/vendor/bundle
           key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          # Cache not found for input keys: Linux-gems-1249f66ba3e6508ebd38170d96559bbd65919b464e8f37bc4ebc5adb9590e0b8, Linux-gems-
```
Comment (rdubigny) [inline .github/workflows/end-to-end.yml:67]: is this comment useful ?

File: cypress/e2e/pass-creation.cy.js [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
   it("lets the user visit the backend domain", () => {
-    cy.visit("https://app-test.moncomptepro.beta.gouv.fr/users/sign-in");
+    cy.visit(
+      "https://app-test.moncomptepro.beta.gouv.fr/users/sign-in"
+    );
   });
 
   it("lets the user fill a FranceConnect form", () => {
     cy.visit("http://localhost:3000/franceconnect");
+    cy.get("form").submit();
```
Comment (rdubigny) [inline cypress/e2e/pass-creation.cy.js:15]: I don't get why we submit the form before filling in the fields ?

File: cypress/support/commands.js [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
+      {
+        method: "POST",
+      }
+    ); // Directly visit the backend login page to avoid CORS issues with the cookies "Same-site: lax" policy
 
     cy.get('input[name="login"]').type(email);
     cy.get("form").submit();
     cy.get('input[name="password"]').type(password);
     cy.get('[action="/users/sign-in"]').submit();
+    cy.setCookie("_session_id", "123key");
```
Comment (rdubigny) [inline cypress/support/commands.js:14]: is this needed ? if the answer is yes you should add a comment to explain.

File: cypress/e2e/pass-creation.cy.js [hunk 4]
```diff
@@ -1,19 +1,18 @@
 describe("DataPass", () => {
-  before(() => {
-    cy.login("user@yopmail.com", "user@yopmail.com");
-  });
-
   beforeEach(() => {
-    Cypress.Cookies.preserveOnce("_session_id");
+    cy.login("user@yopmail.com", "user@yopmail.com");
```
Comment (rdubigny) [inline cypress/e2e/pass-creation.cy.js:3]: does this mean that we should login before each test ? is there no way to do it once for all tests ?

**Review by rdubigny — CHANGES_REQUESTED** [review]
There are several unexpected linting changes. The file seems already linted on master. You should revert linting changes from this PR and update your linting config accordingly.

File: .github/workflows/end-to-end.yml (hunk 1 above)
Comment (Isalafont) [reply .github/workflows/end-to-end.yml:67]: Not anymore since the update of action/cache ! 
Thanks I forget to delete it !

**Review by rdubigny — APPROVED** [review]

**Pull request merged**